examples/views/wait_for_button_press.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+import discord
+from discord.ui import Button, View
+
+
+bot = discord.Bot()
+
+
+@bot.slash_command(guild_ids=[...], description="Wait for button press example from pycord") 
+async def button(ctx):
+    ###############
+    # Create button and add it to View()
+    button = Button(style=discord.ButtonStyle.green, label="Click me!")
+    view = View()
+    view.add_item(button)
+    ###############
+    
+    ###############
+    # Create embed and send message with button and embed.
+    embed = discord.Embed(description="Wait for button press example")
+    embed.set_author(name="Pycord-Development")
+    
+    await ctx.respond(embed=embed, view=view)
+    msg = await ctx.interaction.original_message()
+    ###############
+    
+    ###############
+    # Check whether the user who pressed the button is the ctx.author and the channel
+    def check(interaction):
+        return interaction.channel.id == ctx.channel.id and interaction.user.id == ctx.author.id
+    ###############
+    try:
+        interaction = await bot.wait_for('interaction', check=check, timeout=30)
+        # Wait for the user to press the button.
+
+        if button.custom_id == interaction.data["custom_id"]:
+            # Checks whether the custom_id of the button we specified above is the same as the one the user pressed.
+            await interaction.response.send_message("Button pressed!")
+
+    # Executes if they button is not pressed after a certain time...
+    except asyncio.TimeoutError:
+        embed = discord.Embed(description="Wait for button press example failed (timeout)")
+        embed.set_author(name="Pycord-Development")
+        await msg.edit(embed=embed)
+
+
+
+bot.run("TOKEN")
```

Comment:
I think that this brings it in line with the other view examples that exist already (I have also changed the check to actually handle checking properly instead of just giving up if the user were to invoke the command again in the same channel).

I know that there have been a lot of reviews on this pull request (I have already done one!) but I think that it is important to actually have examples that
a. work,
b. demonstrate a meaningful area of the api and
c. have some sort of practical use.

The examples should also be mildly consistent with one another and likely should follow [PEP-8](https://www.python.org/dev/peps/pep-0008/)'s guidelines on how you should format your program. Therefore I will suggest the following changes (not really broken up because I change quite a lot).

(Sorry for people who will have to review this again)

```suggestion
# If you want a more intuitive way of using message components, check
# out the other examples in this directory

import asyncio

import discord
from discord.ext import commands
from discord import ui


class ButtonBot(commands.Bot):
    def __init__(self):
        super().__init__(command_prefix=commands.when_mentioned_or("$"))

    async def on_ready(self):
        print(f"Logged in as {self.user} (ID: {self.user.id})")
        print("------")


bot = ButtonBot()


@bot.command(name="button", description="Wait for button press example from pycord")
async def button(ctx):
    """Waits for a button press"""
    
    button = ui.Button(style=discord.ButtonStyle.primary, label="Click me!")
    view = ui.View()
    view.add_item(button)

    await ctx.respond("Wait for button press example", view=view)
    message = await ctx.interaction.original_message()
    # ctx.respond doesn't return the message sent so we can get it
    # ourselves

    # Check for whether the interaction is related to this button or not
    def check(interaction):
        return (
            interaction.is_component()
            and button.custom_id == interaction.data["custom_id"]
        )
        # custom_id is a unique identifier for every message component

    try:
        interaction = await bot.wait_for("interaction", check=check, timeout=30)
        # Wait for the user to press the button.

        await interaction.response.send_message("Button pressed!")

    # TimeoutError will be raised by wait_for if an interaction that passes
    # the check is not recieved within 30 seconds
    except asyncio.TimeoutError:
        await message.edit(content="You did not respond after 30 seconds...", view=None)


bot.run("token")
```